src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/ios-helix-job-payload.sh

```diff
@@ -126,7 +126,12 @@ test_results=`ls "$output_directory"/xunit-*.xml`
 if [ ! -f "$test_results" ]; then
     echo "Failed to find xUnit tests results in the output directory. Existing files:"
     ls -la "$output_directory"
-    exit 1
+
+    if [ "$exit_code" == "0" ]; then
```

Comment:
nit, use numeric compare:

```suggestion
    if [ $exit_code -eq 0 ]; then
```